driver/exception.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "driver/platform/platform.h"
+
+#include <stdexcept>
+#include <string>
+
+class SqlException
+    : public std::runtime_error
+{
+public:
+    explicit SqlException(const std::string & message_, const std::string & sql_state_ = "HY000", SQLRETURN return_code_ = SQL_ERROR);
+    const std::string & getSQLState() const noexcept;
+    const SQLRETURN getReturnCode() const noexcept;
```

Comment:
`SqlException`s are used to terminate the flow of execution of the ODBC call AND to fill some info into the corresponding handle's diagnostics record, AND to specify the exit code, the outcome of the call (since it cannot be propagated further). Alternative to this would be a very dirty leaky-abstraction code.